gradle/rewrite.gradle

```diff
@@ -0,0 +1,45 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+project.apply plugin: "org.openrewrite.rewrite"
+
+rewrite {
+  activeRecipe("org.opensearch.openrewrite.DefinitionOfDone")
+  exclusion("**SearchAfterIT.java")
+  exclusion("**StarTreeMapper.java")
+  setExportDatatables(true)
+  setFailOnDryRunResults(true)
+}
+
+dependencies {
+  rewrite("org.openrewrite.recipe:rewrite-static-analysis:2.15.0")
+}
+
+tasks.check.dependsOn(rewriteDryRun)
```

Comment:
i think its not possible to run this no default on dev machine.

Rewrite run should be the roundup solution to fix all the issues breaking the build/quality gate, just like already achieved with spotless. 


PMD is fast enough to run locally on the other hand giving quick indication for broad errors, as some are in common. Other details recipes can not be covered of course, given the need to rewrite to run (in cloud).